graphbolt/src/csc_sampling_graph.cc

```diff
@@ -256,8 +257,76 @@ inline torch::Tensor UniformPick(
       picked_neighbors =
           torch::randint(offset, offset + num_neighbors, {fanout}, options);
     } else {
-      picked_neighbors = torch::randperm(num_neighbors, options);
-      picked_neighbors = picked_neighbors.slice(0, 0, fanout) + offset;
+      picked_neighbors = torch::empty({fanout}, options);
+      AT_DISPATCH_INTEGRAL_TYPES(
+          picked_neighbors.scalar_type(), "UniformPick", ([&] {
+            scalar_t* picked_neighbors_data =
+                picked_neighbors.data_ptr<scalar_t>();
+            if (fanout < num_neighbors / 10) {
+              // If set of numbers is small (up to 64) use linear search to
+              // verify uniqueness this operation is cheaper for CPU.
+              if (fanout && fanout < 64) {
```

Comment:
I read the whole code, first py-code constrain that `fanout >= -1`.

In this case, if `fanout == 0`, `begin` will never equal to `end`. So we add `fanout &&` to avoid this bug. We let the algorithm of the hash-set part deal with this case (fanout == 0).

Reply:
I see, plz leave a comment about the fix such that ppl will be feel confusion in the future when read the code.

Reply:
got